starknet_py/net/udc_deployer/deployer.py

```diff
@@ -86,36 +118,15 @@ async def prepare_contract_deployment(
             calldata=calldata,
         )
 
-        transaction = await self.account.sign_invoke_transaction(
-            calls=call, max_fee=max_fee, auto_estimate=auto_estimate
+        deployer_address = self.deployer_address if self.unique else 0
+        salt = pedersen_hash(self.account.address, salt) if self.unique else salt
```

Comment:
Why is hash calculated here?

Reply:
Because it is calculated inside the UDC contract. To get the same address I have to do the same transformations